rxbroadcast/src/main/java/com/cantrowitz/rxbroadcast/BroadcastProvider.java

```diff
@@ -32,29 +31,28 @@ public void call(final Subscriber<? super Intent> subscriber) {
         final Subscription subscription = Subscriptions.create(new Action0() {
             @Override
             public void call() {
-                unregisterBroadcastReceiver(broadcastReceiver);
+                broadcastRegistrarStrategy.unregisterBroadcastReceiver(broadcastReceiver);
             }
         });
 
-        registerBroadcastReceiver(broadcastReceiver);
+        broadcastRegistrarStrategy.registerBroadcastReceiver(broadcastReceiver);
         subscriber.add(subscription);
     }
 
     @VisibleForTesting
-    BroadcastReceiver getBroadcastReceiver(){
+    BroadcastReceiver getBroadcastReceiver() {
         return broadcastReceiver;
     }
 
-    protected abstract void registerBroadcastReceiver(BroadcastReceiver broadcastReceiver);
-
-    protected abstract void unregisterBroadcastReceiver(BroadcastReceiver broadcastReceiver);
 
     @NonNull
     private BroadcastReceiver createBroadcastReceiver(final Subscriber<? super Intent> subscriber) {
```

Comment:
I'm down with that. I just figured the reason for it was the previous subclassing but I guess not.